### PR TITLE
[CI] Add test owners

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
This PR adds "team:{serve,core,ml,autoscaler,ant-group}" tags to py_test and cc_test bazel targets. This is used to facilitate fine-grained tracking of test pass rate.

### CODE REVIEW

**Review by richardliaw — COMMENTED** [review]
Looks good from my side. FYI @amogkam @krfricke @sven1977 .

**Comment by edoakes** [comment]
@simon-mo what is the plan for tagging windows separately here?

**Comment by simon-mo** [comment]
@edoakes Windows is orthogonal. We already have metadata associated with the testing OS through another pipeline and it will be processed in a separate step.

**Review by edoakes — APPROVED** [review]

File: python/ray/tests/BUILD [hunk 1]
```diff
@@ -299,7 +302,7 @@ py_test_module_list(
   # TODO(barakmich): py_test will support env in Bazel 4.0.0...
   # Until then, we can use tags.
   #env = {"RAY_CLIENT_MODE": "1", "RAY_PROFILING": "1"},
-  tags = ["exclusive", "client_tests"],
+  tags = ["exclusive", "client_tests", "team:core"],
```
Comment (ericl) [inline python/ray/tests/BUILD:305]: I believe client is still owned by serverless (cc @AmeerHajAli )

File: cpp/BUILD.bazel [hunk 2]
```diff
@@ -159,6 +160,7 @@ cc_test(
         "ray_cpp_pkg",
     ],
     linkstatic = True,
+    tags = ["team:core"],
```
Comment (ericl) [inline cpp/BUILD.bazel:163]: C++ package is owned by serverless atm.

File: python/ray/util/dask/BUILD [hunk 3]
```diff
@@ -16,15 +16,15 @@ py_test(
     size = "small",
     main = "test_dask_callback.py",
     srcs = ["tests/test_dask_callback.py"],
-    tags = ["exclusive", "client"],
+    tags = ["team:core", "exclusive", "client"],
```
Comment (ericl) [inline python/ray/util/dask/BUILD:19]: .

Comment (ericl) [inline python/ray/util/dask/BUILD:19]: Client tests => serverless

File: python/ray/util/dask/BUILD [hunk 4]
```diff
@@ -33,15 +33,15 @@ py_test(
     size = "small",
     main = "test_dask_optimization.py",
     srcs = ["tests/test_dask_optimization.py"],
-    tags = ["exclusive", "client"],
+    tags = ["team:core", "exclusive", "client"],
```
Comment (ericl) [inline python/ray/util/dask/BUILD:36]: Client tests => serverless

File: python/ray/util/dask/BUILD [hunk 5]
```diff
@@ -72,15 +72,15 @@ py_test(
     size = "medium",
     main = "dask_ray_persist_example.py",
     srcs = ["examples/dask_ray_persist_example.py"],
-    tags = ["exclusive", "client"],
+    tags = ["team:core", "exclusive", "client"],
```
Comment (ericl) [inline python/ray/util/dask/BUILD:75]: Client tests => serverless

File: python/ray/util/dask/BUILD [hunk 6]
```diff
@@ -89,15 +89,15 @@ py_test(
     size = "medium",
     main = "dask_ray_scheduler_example.py",
     srcs = ["examples/dask_ray_scheduler_example.py"],
-    tags = ["exclusive", "client"],
+    tags = ["team:core", "exclusive", "client"],
```
Comment (ericl) [inline python/ray/util/dask/BUILD:92]: Client tests => serverless

File: python/ray/util/dask/BUILD [hunk 7]
```diff
@@ -107,7 +107,7 @@ py_test(
     size = "medium",
     main = "dask_ray_shuffle_optimization.py",
     srcs = ["examples/dask_ray_shuffle_optimization.py"],
-    tags = ["exclusive", "client"],
+    tags = ["team:core", "exclusive", "client"],
```
Comment (ericl) [inline python/ray/util/dask/BUILD:110]: Client tests => serverless

**Review by ericl — APPROVED** [review]
LGTM to merge as is. I suggested some owner changes for client tests, but it is minor.

**Pull request merged**